### PR TITLE
Support for Archival Resource Key (ARK) identifiers

### PR DESCRIPTION
**Pull request by datadavev**
Support for Archival Resource Key (ARK) identifiers

### CODE REVIEW

**Comment by davidlehn** [comment]
Is this related to https://github.com/perma-id/w3id.org/pull/3237?
Out of curiosity, is this using `arkid` due to `ARK` already existing?

File: arkid/.htaccess [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# Any ARK value resolution
+# e.g. https://w3id.org/arkid/12345/foo
+RewriteRule ^([0-9]{4,8})/(.*)$ https://n2t.net/ark:$1/$2 [R=302,L]
+
+# NAAN registry in ANVL
+RewriteRule ^naans?|registry$ https://n2t.net/e/pub/naan_registry.txt [R=302,L]
+
+# Register or update NAAN
+RewriteRule ^naans?|register$ http://n2t.net/e/naan_request [R=302,L]
```
Comment (davidlehn) [inline arkid/.htaccess:20]: The previous rule will catch `nanns?`.  Should this be just `register`?

**Comment by datadavev** [comment]
> Is this related to #3237? Out of curiosity, is this using `arkid` due to `ARK` already existing?

Yes, correct. Using "ark" would be preferable, but likely confusing with the existing "ARK" entry.

File: arkid/.htaccess (hunk 1 above)
Comment (datadavev) [reply arkid/.htaccess:20]: Yes, copy n' paste slop. I'll correct it. Thanks.

**Comment by davidlehn** [comment]
> Yes, correct. Using "ark" would be preferable, but likely confusing with the existing "ARK" entry.

I'm not sure of the state of the world now, but in the past I think some file systems just overwrote files that only differed in case.  Want to try and play nice with that.

You could ask the ARK maintainers if they are still using that prefix.  I imagine some projects become obsolete over time, but I didn't look into their work.  Of course "cool URIs don't change", so taking over anything would need to be done with care.

**Comment by datadavev** [comment]
> I'm not sure of the state of the world now, but in the past I think some file systems just overwrote files that only differed in case. Want to try and play nice with that.

There is one instance of this problem in w3id.org now, `airo` and `AIRO` results in a very unpleasant experience with git on my mac, which happens to be formatted case aware, but not sensitive.

**Comment by davidlehn** [comment]
Oops.  Yeah, that slipped through and has an active PR.  I will try to address it.  Future plan is to have an automated check for that issue.

**Pull request merged**